pkg/kclient/pods.go

```diff
@@ -40,8 +39,12 @@ func (c *Client) WaitAndGetPodWithEvents(selector string, desiredPhase corev1.Po
 
 	klog.V(3).Infof("Waiting for %s pod", selector)
 
-	spinner := log.Spinner(waitMessage)
-	defer spinner.End(false)
+	var spinner *log.Status
+	defer func() {
+		if spinner != nil {
```

Comment:
In what situation will this check be not satisfied? We're doing `var spinner *log.Status` right above this so, I'm confused as to when `spinner` will be `nil` and this check would be not satisfied.

Reply:
In `var spinner *log.Status`, spinner is a non initialized pointer, so getting its zero value, nil. 
The pointer gets affected a value only when receiving a phase different from the expected one.